R/get_orcp_cn.R

```diff
@@ -570,3 +570,31 @@ line_merge = function(cohesive_network_city_boundary, OS_combined_net_city_bound
 
   return(os_buffer_NPT_group_combined)
 }
+
+
+convert_to_linestrings = function(geometries) {
+  # Check the geometry types
+  geometry_types = sf::st_geometry_type(geometries)
+  
+  # If the unique geometry types are not just LINESTRING, perform the conversion
+  if (!all(unique(geometry_types) == "LINESTRING")) {
+    
+    # Filter out LINESTRING geometries
+    linestrings = geometries[geometry_types == "LINESTRING", ]
+    
+    # Filter out MULTILINESTRING geometries and cast them to LINESTRING
+    multilinestrings = geometries[geometry_types == "MULTILINESTRING", ]
+    multilinestrings = sf::st_cast(multilinestrings, "LINESTRING")
+    
+    # Combine the LINESTRING and converted MULTILINESTRING geometries
+    combined = rbind(linestrings, multilinestrings)
```

Comment:
Have you tried a different function, e.g. `dplyr::bind_rows()` here? May be worth trying if this is called frequently or if it's used on large datasets.